src/main/webapp/map.js

```diff
@@ -120,14 +144,54 @@ function createMarker(place) {
 }
 
 function displayMarkers() {
-  for (var i = 0; i < markers.length; i++ ) {
-    markers[i].setMap(map);
-  }
+  markers.forEach(marker => marker.setMap(map));
 }
 
 function clearMarkers() {
-  for (var i = 0; i < markers.length; i++ ) {
-    markers[i].setMap(null);
-  }
+  markers.forEach(marker => marker.setMap(null));
   markers = [];
 }
+
+/**
+/* Calculates the straight line distance between two
+/* sets of latitutes and longitutes
+/* A breakdown of the haversine formula can be found at
+/* http://www.movable-type.co.uk/scripts/latlong.html
+ */
+function haversine_distance(lat1,lon1,lat2,lon2) {
```

Comment:
Good work on implementing this. Consider adding tests to verify this algo returns reasonable results. e.g.
1) if the same location is passed in twice, is the result ~zero?
2) If you pass in two locations that you know to be 1000 metres apart, what does your algorithm's implementation return?